### PR TITLE
[OFFAPPS-1037] New scaffold documentations

### PR DESCRIPTION
**Pull request by yu-tian113**
[OFFAPPS-1037] Replace master branch with offapps-migration branch. Create new readme documentation and ADR

## Description
This PR is part of ticket [OFFAPPS-1037](https://zendesk.atlassian.net/browse/OFFAPPS-1037), including changes and additions to the new scaffold documentation

## Tasks
- [x] Update README.md
- [x] Update i18n.md
- [x] Add DEPLOY.md
- [x] Add ADR for replacing the old scaffolding with new one

## References
[JIRA ticket](https://zendesk.atlassian.net/browse/OFFAPPS-1037)

@zendesk/apps-migration
@zendesk/vegemite 

## Risks

### CODE REVIEW

**Comment by keitheous** [comment]
👍

File: doc/adrs/replace-migration-scaffold-with-clean-scaffold-in-master-branch.md [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+### New app scaffold
+A new *clean* app scaffold has been build on branch [offapps-migration](https://github.com/zendesk/app_scaffold/tree/offapps-migration) based on branch [from-scratch](https://github.com/zendesk/app_scaffold/tree/from-scratch)
+
+### Move migration scaffold into App Migrator repository
+Migration scaffold in master branch will be moved into [App Migrator](https://github.com/zendesk/zendesk_app_migrator), used and maintained as *built-in* template resource
+
+### Move new app scaffold to master branch
+
+### Add support in [ZAT](https://github.com/zendesk/zendesk_apps_tools) to create new app with the new scaffold
+* New command option `zat new scaffold` will create a v2 app using the new app scaffold.
```
Comment (ocke) [inline doc/adrs/replace-migration-scaffold-with-clean-scaffold-in-master-branch.md]: not `--scaffold`?

Comment (vaevictis) [reply doc/adrs/replace-migration-scaffold-with-clean-scaffold-in-master-branch.md]: I don't know what's the standard in `zat`'s interface, but I'd stick to whatever it is.
If `scaffold` is supposed to be a sub-command, and there are other sub-commands, then this should be fine. If everything past the first subcommand is a flag, then we should keep the tool's interface consistant.

File: README.md [hunk 2]
```diff
@@ -8,143 +8,107 @@ This repo contains a scaffold to help developers build [apps for Zendesk product
...
 
 #### spec
 The spec directory is where all your tests and test helpers live. Tests are not required to submit/upload your app to Zendesk and your test files are not included in your app's package, however it is good practice to write tests to document functionality and prevent bugs.
 
 #### src
-The src directory is where your raw source code lives. The App Scaffold includes different directories for JavaScript, stylesheets, templates and translations. Most of your additions will be in here (and spec, of course!).
+The src directory is where your raw source code lives. The App Scaffold includes different directories for JavaScript, stylesheets, templates, images and translations. Most of your additions will be in here (and spec, of course!).
+
+#### webpack
+In this directory, we have the translations-loader.js which is used by webpack to convert translation json files to javascript objects. We also have the translations-plugin.js to extract multilingual project meta data from translation json files for Zendesk Market.
```
Comment (vaevictis) [inline README.md]: This folder contains custom tooling to process translations at build time:
- `translations-loader.js` is used by Webpack to convert `.json` translation files to JavaScript objects, for the app itself.
- `translations-plugin.js` is used to extract compulsory translation strings from the `en.json` file that are used to display metadata about your app on the Zendesk Marketplace.

File: README.md [hunk 3]
```diff
@@ -8,143 +8,107 @@ This repo contains a scaffold to help developers build [apps for Zendesk product
...
+
+#### .babelrc
+[.babelrc](https://babeljs.io/docs/en/babelrc.html) is the configuration file for babel compiler.
 
-#### .eslintrc
-.eslintrc is a configuration file for [ESLint](http://eslint.org). ESLint is a linting utility for JavaScript. For more information on how to configure ESLint, see [Configuring ESLint](http://eslint.org/docs/user-guide/configuring).
+#### .browserslistrc
+.browserslistrc is a configuration file to specify supported browsers of your application, some develop/build tools read info from this file if it exists in your project root. At present, our scaffolding doesn't reply on this file, [default browserslist query](https://github.com/browserslist/browserslist#queries) is used by [Babel](https://babeljs.io/docs/en/babel-preset-env/) and [PostCSS](https://github.com/csstools/postcss-preset-env#browsers)
```
Comment (vaevictis) [inline README.md]: supported browsers of your application =>  browsers supported by your application

File: README.md [hunk 4]
```diff
@@ -8,143 +8,107 @@ This repo contains a scaffold to help developers build [apps for Zendesk product
...
 
-#### karma.conf.js
-karma.conf.js is a configuration file for [Karma](http://karma-runner.github.io). Karma is a JavaScript test runner. This file defines where your source and test files live. For more information on how to use this file, see [Karma - Configuration File](http://karma-runner.github.io/1.0/config/configuration-file.html).
+#### jest.config.js
+[jest.config.js](https://jestjs.io/docs/en/configuration.html) is the configuration file for Jest
 
 #### package.json
-package.json is a configuration file for [NPM](https://www.npmjs.com). NPM is a package manager for JavaScript. This file includes information about your project and its dependencies. For more information on how to configure this file, see [package.json](https://docs.npmjs.com/files/package.json).
+package.json is the configuration file for [Yarn](https://yarnpkg.com/), which is a package managers for JavaScript. This file includes information about your project and its dependencies. For more information on how to configure this file, see [Yarn package.json](https://yarnpkg.com/en/docs/package-json).
```
Comment (vaevictis) [inline README.md]: singular `manager`

File: README.md [hunk 5]
```diff
@@ -210,7 +172,10 @@ taking note of the created filename.
 For more information on the Zendesk Apps Tools please see the [documentation](https://developer.zendesk.com/apps/docs/apps-v2/getting_started#zendesk-app-tools).
 
 ## External Dependencies
-External dependencies are defined in a module, [`lib/external_assets.js`](https://github.com/zendesk/app_scaffold/blob/master/lib/external_assets.js). The export of the module is imported into [`webpack.config.js`](https://github.com/zendesk/app_scaffold/blob/master/webpack.config.js) at build-time. This ensures these dependencies are included on your app's `index.html` as well as in the test suite.
+External dependencies are defined in [webpack.config.js](https://github.com/zendesk/app_scaffold/blob/master/webpack.config.js#L22). This ensures these dependencies are included on your app's `index.html`.
```
Comment (vaevictis) [inline README.md]: **in** your app's (...)

File: README.md [hunk 6]
```diff
@@ -210,7 +172,10 @@ taking note of the created filename.
...
+
+## Zendesk Garden CSS Components
+Included [Zendesk Garden CSS Components](https://garden.zendesk.com/css-components/) are listed in [package.json](https://github.com/zendesk/app_scaffold/blob/master/package.json#L25) as dev dependencies. Instead of importing them into the app css bundle, we are building a [jsDelivr CDN](https://www.jsdelivr.com/) link from the dependencies list and inject the link into `index.html` as another `<style>` tag. Feel free to add/remove the Garden components as need, webpack will generate and insert the updated link during the build process.
```
Comment (vaevictis) [inline README.md]: as need**ed**

Also, I didn't know about this jsDelivr approach. Is it new? Or did I completely miss something?

File: doc/adrs/replace-migration-scaffold-with-clean-scaffold-in-master-branch.md [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+title: "ADR: Replace migration Scaffold with clean Scaffold in Master Branch"
+date: 2018-08-16
+draft: false
+status: "Accepted"
+---
+
+# ADR: Replace migration Scaffold with clean Scaffold in Master Branch
+
+## Context
+At the moment, the master branch has the scaffold specifically build for helping developers to migrate v1 apps to v2. From monitor, the majority usage comes from [App Migrator](https://github.com/zendesk/zendesk_app_migrator) and [ZAT](https://github.com/zendesk/zendesk_apps_tools) which use the master branch as the download source.
```
Comment (vaevictis) [inline doc/adrs/replace-migration-scaffold-with-clean-scaffold-in-master-branch.md]: I believe it should be _built for + NAME_, like built for speed, or in this case _built to help_. Then ditch **to** migrate.
> the master branch has the scaffold specifically built to help developers migrate v1 apps to v2.

File: doc/adrs/replace-migration-scaffold-with-clean-scaffold-in-master-branch.md [hunk 8]
```diff
@@ -0,0 +1,34 @@
...
+It should be more helpful to show a *clean*(from scratch) scaffold to the developers directly come to this repo for a starting point to build apps
```
Comment (vaevictis) [inline doc/adrs/replace-migration-scaffold-with-clean-scaffold-in-master-branch.md]: Don't be scared by your opinions :)
> It **is** more helpful to show developers coming directly to this repo a *clean* (from scratch) scaffold, one usable as a starting point to build new apps respecting our current conventions out of the box.

File: doc/adrs/replace-migration-scaffold-with-clean-scaffold-in-master-branch.md [hunk 9]
```diff
@@ -0,0 +1,34 @@
...
+
+## Decision
+
+### New app scaffold
+A new *clean* app scaffold has been build on branch [offapps-migration](https://github.com/zendesk/app_scaffold/tree/offapps-migration) based on branch [from-scratch](https://github.com/zendesk/app_scaffold/tree/from-scratch)
```
Comment (vaevictis) [inline doc/adrs/replace-migration-scaffold-with-clean-scaffold-in-master-branch.md]: buil**t**

File: doc/adrs/replace-migration-scaffold-with-clean-scaffold-in-master-branch.md [hunk 10]
```diff
@@ -0,0 +1,34 @@
...
+
+### Move migration scaffold into App Migrator repository
+Migration scaffold in master branch will be moved into [App Migrator](https://github.com/zendesk/zendesk_app_migrator), used and maintained as *built-in* template resource
```
Comment (vaevictis) [inline doc/adrs/replace-migration-scaffold-with-clean-scaffold-in-master-branch.md]: as **a** *built-in* template resource

File: doc/adrs/replace-migration-scaffold-with-clean-scaffold-in-master-branch.md (hunk 1 above)
Comment (yu-tian113) [reply doc/adrs/replace-migration-scaffold-with-clean-scaffold-in-master-branch.md]: good catch, miss it. Should be option --scaffold

File: README.md [hunk 11]
```diff
@@ -8,143 +8,107 @@ This repo contains a scaffold to help developers build [apps for Zendesk product
...
+- [PostCSS](https://postcss.org//) stylesheets
 
-Webpack compiles web browser applications. It allows splitting your source code into modules and re-use them with require and import statements. It also allows splitting your compiled project into separate files that are loaded on demand.
+PostCSS transforms stylesheets with JS plugins. These plugins can lint your CSS, support variables and mixins, transpile future CSS syntax, inline images, and more.
 
-- [Karma](http://karma-runner.github.io/) test runner
+- [standardJS](https://standardjs.com/) JS linting
 
-The main goal for Karma is to bring a productive testing environment to developers with minimal configuration.
+standardJS is a javascript style guide, it helps catching style issues or code errors, and automatically format code for you.
```
[the remark this answers is not in the document]
Comment (yu-tian113) [reply README.md]: 👍

File: README.md [hunk 12]
```diff
@@ -8,143 +8,107 @@ This repo contains a scaffold to help developers build [apps for Zendesk product
...
 
-- [Jasmine](https://jasmine.github.io/) testing framework
+- [Jest](https://jestjs.io/) Javascript testing framework
 
-Jasmine is a behavior-driven development framework for testing JavaScript code with a clean syntax.
+Jest is bundled with JSDom and built on top of Jasmine. It's more than just a ReactJS testing framework. At Zendesk Apps team, we use it for unit testing and integration testing on our official apps.
```
[the remark this answers is not in the document]
Comment (yu-tian113) [reply README.md]: 👍

File: README.md [hunk 13]
```diff
@@ -8,143 +8,107 @@ This repo contains a scaffold to help developers build [apps for Zendesk product
...
 | [`webpack.config.js`](#webpackconfigjs) | Configuration file that webpack uses to build your app                                       |
 
 #### dist
-The dist directory is the folder you will need to package when submitting your app to the marketplace. It is also the folder you will have to serve when using [ZAT](https://developer.zendesk.com/apps/docs/apps-v2/getting_started#zendesk-app-tools). It includes your app's manifest.json file, an assets folder with all your compiled JavaScript and CSS as well as HTML and images.
-
-#### lib
-The lib directory is where the source code for the app shims and compatibility methods live. While you may modify or remove this code as required for your app, doing so is not recommended for beginners.
+The dist directory is created when you run the app building scripts, it is the folder you will need to package when submitting your app to the marketplace. It is also the folder you will have to serve when using [ZAT](https://developer.zendesk.com/apps/docs/apps-v2/getting_started#zendesk-app-tools). It includes your app's manifest.json file, an assets folder with all your compiled JavaScript and CSS as well as HTML and images.
```
[the remark this answers is not in the document]
Comment (yu-tian113) [reply README.md]: 👍

File: README.md (hunk 2 above)
Comment (yu-tian113) [reply README.md]: `translations-plugin.js` is used to extract compulsory translation strings from the language json files that are used to display metadata about your app on the Zendesk Marketplace.

File: README.md (hunk 3 above)
Comment (yu-tian113) [reply README.md]: 👍

File: README.md (hunk 4 above)
Comment (yu-tian113) [reply README.md]: 👍

File: README.md (hunk 5 above)
Comment (yu-tian113) [reply README.md]: 👍

File: doc/adrs/replace-migration-scaffold-with-clean-scaffold-in-master-branch.md (hunk 7 above)
Comment (yu-tian113) [reply doc/adrs/replace-migration-scaffold-with-clean-scaffold-in-master-branch.md]: 👍

File: doc/adrs/replace-migration-scaffold-with-clean-scaffold-in-master-branch.md (hunk 8 above)
Comment (yu-tian113) [reply doc/adrs/replace-migration-scaffold-with-clean-scaffold-in-master-branch.md]: 👍

File: doc/adrs/replace-migration-scaffold-with-clean-scaffold-in-master-branch.md (hunk 9 above)
Comment (yu-tian113) [reply doc/adrs/replace-migration-scaffold-with-clean-scaffold-in-master-branch.md]: 👍

File: doc/adrs/replace-migration-scaffold-with-clean-scaffold-in-master-branch.md (hunk 10 above)
Comment (yu-tian113) [reply doc/adrs/replace-migration-scaffold-with-clean-scaffold-in-master-branch.md]: 👍

File: README.md [hunk 14]
```diff
@@ -8,143 +8,107 @@ This repo contains a scaffold to help developers build [apps for Zendesk product
...
+- [Zendesk Garden](https://garden.zendesk.com/css-components/) UI components
 
-Handlebars is a powerful templating library that lets you build semantic templates for your app with minimal logic.
+Collection of user interface components for Zendesk products. You’ll find components built to respond to a range of user input devices, tuned to handle right-to-left layouts, and finessed with just the right touch of subtle animation.
 
-- [SASS](http://sass-lang.com/) stylesheets
+- [Webpack 4](https://webpack.github.io/) module bundler
 
-Sass is an extension of CSS that adds power and elegance to the basic language. It allows you to use variables, nested rules, mixins, inline imports, and more.
+Webpack compiles web browser applications. It allows splitting your source code into modules and re-use them with require and import statements. It also allows splitting your compiled project into separate files that are loaded on demand.
```
[the remark this answers is not in the document]
Comment (yu-tian113) [reply README.md]: Yeah, rewrote with only a brief introduction of webpack

File: README.md [hunk 15]
```diff
@@ -8,143 +8,110 @@ This repo contains a scaffold to help developers build [apps for Zendesk product
...
 
-#### .eslintrc
-.eslintrc is a configuration file for [ESLint](http://eslint.org). ESLint is a linting utility for JavaScript. For more information on how to configure ESLint, see [Configuring ESLint](http://eslint.org/docs/user-guide/configuring).
+#### webpack
+This directory contains custom tooling to process translations at build time:
 
-#### karma.conf.js
-karma.conf.js is a configuration file for [Karma](http://karma-runner.github.io). Karma is a JavaScript test runner. This file defines where your source and test files live. For more information on how to use this file, see [Karma - Configuration File](http://karma-runner.github.io/1.0/config/configuration-file.html).
+- translations-loader.js is used by Webpack to convert .json translation files to JavaScript objects, for the app itself.
+- translations-plugin.js is used to extract compulsory translation strings from the language json files that are used to display metadata about your app on the Zendesk Marketplace.
```
Comment (hoangtrvu) [inline README.md]: should be `Zendesk Apps Marketplace` to be exact

File: doc/adrs/replace-migration-scaffold-with-clean-scaffold-in-master-branch.md [hunk 16]
```diff
@@ -0,0 +1,34 @@
...
+title: "ADR: Replace migration Scaffold with clean Scaffold in Master Branch"
+date: 2018-08-16
+draft: false
+status: "Accepted"
+---
+
+# ADR: Replace migration Scaffold with clean Scaffold in Master Branch
+
+## Context
+At the moment, the master branch has the scaffold specifically build to help developers migrate v1 apps to v2. From monitor, the majority usage comes from [App Migrator](https://github.com/zendesk/zendesk_app_migrator) and [ZAT](https://github.com/zendesk/zendesk_apps_tools) which use the master branch as the download source.
```
Comment (hoangtrvu) [inline doc/adrs/replace-migration-scaffold-with-clean-scaffold-in-master-branch.md]: `specifically built`?

File: DEPLOY.md [hunk 17]
```diff
@@ -0,0 +1,17 @@
+Note: Run commands in the root app directory.
+
+Compile the app for DEV
+===============
+1) `yarn install`
+2) `yarn run watch`
+3) `zat server -p dist` - Serves the app to your zendesk instance with `?zat=true`
```
[the remark this answers is not in the document]
Comment (yu-tian113) [reply DEPLOY.md]: 👍

[the remark this answers is not in the document]
Comment (yu-tian113) [reply DEPLOY.md]: 👍

File: README.md [hunk 18]
```diff
@@ -8,143 +8,110 @@ This repo contains a scaffold to help developers build [apps for Zendesk product
...
-_Note: The App Scaffold currently depends on zat v1.35.12 or greater._
-
-Foreman allows you to easily run multiple processes in one tab. One process is `zat server --path=./dist`, which serves the app in a way that can be run in a supported Zendesk product. The second is `webpack --watch` to rebuild the project whenever you save changes to a source file.
-
-To run these processes, run
+To serve the app to your zendesk instance with `?zat=true`, run
 
 ```
-nf start
+yarn run build:dev
```
[the remark this answers is not in the document]
Comment (yu-tian113) [reply README.md]: 👍

File: README.md [hunk 19]
```diff
@@ -8,143 +8,110 @@ This repo contains a scaffold to help developers build [apps for Zendesk product
...
+Webpack is a module bundler, we use it to bundle up javascript modules for use as web applications, also to perform tasks like transforming and transpiling, etc.
 
-- [Webpack](https://webpack.github.io/) module bundler
+- [PostCSS](https://postcss.org//) stylesheets
 
-Webpack compiles web browser applications. It allows splitting your source code into modules and re-use them with require and import statements. It also allows splitting your compiled project into separate files that are loaded on demand.
+PostCSS transforms stylesheets with JS plugins. These plugins can lint your CSS, support variables and mixins, transpile future CSS syntax, inline images, and more.
 
-- [Karma](http://karma-runner.github.io/) test runner
+- [standardJS](https://standardjs.com/) JS linting
```
[the remark this answers is not in the document]
Comment (yu-tian113) [reply README.md]: 👍

File: README.md [hunk 20]
```diff
@@ -8,143 +8,110 @@ This repo contains a scaffold to help developers build [apps for Zendesk product
...
 
-The main goal for Karma is to bring a productive testing environment to developers with minimal configuration.
+standardJS is a javascript style guide, it helps catching style issues or code errors, and automatically formats code for you.
```
[the remark this answers is not in the document]
Comment (yu-tian113) [reply README.md]: 👍

File: README.md (hunk 15 above)
Comment (yu-tian113) [reply README.md]: 👍

File: doc/adrs/replace-migration-scaffold-with-clean-scaffold-in-master-branch.md (hunk 16 above)
Comment (yu-tian113) [reply doc/adrs/replace-migration-scaffold-with-clean-scaffold-in-master-branch.md]: 👍

File: README.md [hunk 21]
```diff
@@ -8,143 +8,110 @@ This repo contains a scaffold to help developers build [apps for Zendesk product
 ## Getting Started
 
 ### Dependencies
-- [Node.js](https://nodejs.org/en/) >= 6.3.x
+- [Node.js](https://nodejs.org/en/) >=8.9.4
```
[the remark this answers is not in the document]
Comment (yu-tian113) [reply README.md]: Set it back to 6.3.x to be in sync with new scaffold

File: README.md [hunk 22]
```diff
@@ -8,143 +8,110 @@ This repo contains a scaffold to help developers build [apps for Zendesk product
 ## Getting Started
```
[the remark this answers is not in the document]
Comment (yu-tian113) [reply README.md]: 👍

File: DEPLOY.md [hunk 23]
```diff
@@ -0,0 +1,18 @@
...
+===============
+1) `yarn install`
+2) `yarn run watch`
+3) Open a new command line window in the root app directory
+4) `zat server -p dist` - Serves the app to your Zendesk instance with `?zat=true`
+
+Compile the app for PROD
+===============
+1) `yarn install --production`
+2) `yarn run build:prod`
```
[the remark this answers is not in the document]
Comment (yu-tian113) [reply DEPLOY.md]: 👍

**Pull request merged**